ui/selectors/multichain.ts

```diff
@@ -212,9 +212,36 @@ export function getMultichainNetwork(
   // on having a non-EVM account being selected!
   const selectedAccount = account ?? getSelectedInternalAccount(state);
   const nonEvmNetworks = getMultichainNetworkProviders(state);
-  const nonEvmNetwork = nonEvmNetworks.find((provider) => {
-    return provider.isAddressCompatible(selectedAccount.address);
-  });
+  const selectedChainId = state.metamask.selectedMultichainNetworkChainId;
+
+  let nonEvmNetwork: MultichainProviderConfig | undefined;
+
+  // First try to find network by selectedChainId
+  if (selectedChainId) {
+    nonEvmNetwork = nonEvmNetworks.find(
+      (provider) => provider.chainId === selectedChainId,
+    );
+  }
+
+  // If no network found by selectedChainId, we try to find by scopes
+  if (!nonEvmNetwork && selectedAccount.scopes?.length > 0) {
```

Comment:
Is the `?` for `scopes` required here? 🤔 It should always be defined on `InternalAccount`s, we made sure it was defined and non-optional, since it's now a "core-thing" for an account